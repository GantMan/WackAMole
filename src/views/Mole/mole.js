import React, { Component } from 'react'
import { View, Image, TouchableWithoutFeedback } from 'react-native'

const hole = require('./img/hole.png')
const mole = require('./img/idle.png')
const hit = require('./img/hit.png')
const struck = require('./img/done.png')

export default class Mole extends Component {

  constructor (props) {
    super(props)
    this.state = {
      ...props,
      currentImage: props.active ? mole : hole
    }
  }

  handlePressIn = () => {
    if (this.props.active) this.setState({currentImage: hit})
  }

  handlePressOut = () => {
    if (this.props.active) this.setState({currentImage: struck})
  }

  render () {
    return (
      <View style={styles.GrassPatch}>
        <TouchableWithoutFeedback
          onPressIn={this.handlePressIn}
          onPressOut={this.handlePressOut}
        >
          <Image source={this.state.currentImage} />
        </TouchableWithoutFeedback>
      </View>
    )
  }
}

const styles = {
  GrassPatch: {
    // Code to stop pixel jerk
    justifyContent: 'flex-end',
    minHeight: 115
  }
}
