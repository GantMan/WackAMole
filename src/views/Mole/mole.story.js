import React from 'react'
import { storiesOf } from '@storybook/react-native'
import Mole from './mole'

// If a mole is started active, when it is tapped it will
// show a struck state, and stop there.
storiesOf('Mole')
  .add('Default Mole', () => (
    <Mole />
  ))
  .add('Active Mole', () => (
    <Mole active />
  ))
