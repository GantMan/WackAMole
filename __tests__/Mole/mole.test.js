import React from 'react'
import Mole from '../../src/views/Mole/mole'
import renderer from 'react-test-renderer'

test('Inactive Mole Press', () => {
  const moleInstance = renderer.create(<Mole />).getInstance()
  // console.log(moleInstance.state)
  moleInstance.state.currentImage.testUri.includes('hole')
  moleInstance.handlePressIn()
  expect(moleInstance.state.currentImage.testUri).toContain('hole')
})

test('Active Mole Press', () => {
  const moleInstance = renderer.create(<Mole active />).getInstance()
  // console.log(moleInstance.state)
  expect(moleInstance.state.currentImage.testUri).toContain('idle')
  moleInstance.handlePressIn()
  expect(moleInstance.state.currentImage.testUri).toContain('hit')
  // Take a picture it'll last longer
  expect(moleInstance.state).toMatchSnapshot()
})

test('Snapshot of default shape', () => {
  const tree = renderer
    .create(<Mole />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('Snapshot of active', () => {
  const tree = renderer
    .create(<Mole active />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
