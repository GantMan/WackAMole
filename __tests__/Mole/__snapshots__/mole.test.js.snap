// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active Mole Press 1`] = `
Object {
  "active": true,
  "currentImage": Object {
    "testUri": "../../../src/views/Mole/img/hit.png",
  },
}
`;

exports[`Snapshot of active 1`] = `
<View
  style={
    Object {
      "justifyContent": "flex-end",
      "minHeight": 115,
    }
  }
>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../src/views/Mole/img/idle.png",
      }
    }
  />
</View>
`;

exports[`Snapshot of default shape 1`] = `
<View
  style={
    Object {
      "justifyContent": "flex-end",
      "minHeight": 115,
    }
  }
>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../src/views/Mole/img/hole.png",
      }
    }
  />
</View>
`;
